src/IDHelper.java

```diff
@@ -1,77 +1,74 @@
-        }
-    }
-
-    private static String chekID(String id){
-        try{
-            char[] symbolArray = id.toCharArray();//ВОПРОС
-            int chek = Integer.parseInt(id);
-            if (symbolArray.length != 8){
-                JOptionPane.showMessageDialog(null, "Ошибка: ID должен состоять из 8 символов","",JOptionPane.PLAIN_MESSAGE);
-                return "no";
+    /**
+     * Validates the given ID string. The method checks if the ID is a valid integer,
+     * has the required length, and exists in the database.
+     *
+     * @param id The ID string to be validated.
+     * @return {@code true} if the ID is valid and exists in the database, {@code false} otherwise.
+     *
+     * @throws NumberFormatException if the ID cannot be parsed as an integer.
+     */
+    public static boolean validateId(String id) {
+        try {
+            // check that ID is a valid Integer
+            Integer.parseInt(id);
+            // check that ID has required length
+            if (id.length() != REQUIRED_ID_LENGTH) {
+                JOptionPane.showMessageDialog(null, "ID должен состоять из 8 символов",
+                        "Ошибка", JOptionPane.PLAIN_MESSAGE);
+                return false;
             }
-            else if (passReader(id) == "nope"){
-                JOptionPane.showMessageDialog(null,"Ваш ID не зарегистрирован","",JOptionPane.PLAIN_MESSAGE);
-                return "no";
+            // check that ID exists in the database
+            else if (!checkIdExistence(id)) {
+                JOptionPane.showMessageDialog(null,"Ваш ID не зарегистрирован",
+                        "Ошибка", JOptionPane.PLAIN_MESSAGE);
+                return false;
             }
-            else{
-                return "yes";
+            else {
+                return true;
             }
-        } catch(NumberFormatException e){
-            JOptionPane.showMessageDialog(null,"Ошибка: ID не может содержать букв и сторонних символов","",JOptionPane.PLAIN_MESSAGE);
-            return "no";
+        } catch (NumberFormatException e) {
+            JOptionPane.showMessageDialog(null, "ID не может содержать букв и сторонних символов",
+                    "Ошибка", JOptionPane.PLAIN_MESSAGE);
+            return false;
         }
     }
 
-    private static String passReader(String id) {
-        boolean chek = false;
+    /**
+     * Checks if the given ID exists in the database stored in the "baza.txt" file.
+     *
+     * @param id The ID string to be checked for existence in the database.
+     * @return {@code true} if the ID exists in the database, {@code false} otherwise.
+     *
+     * @throws RuntimeException if an IOException occurs while reading the database file.
+     */
+    private static boolean checkIdExistence(String id) {
+        boolean success = false;
         try {
-            BufferedReader baza = new BufferedReader(new FileReader("/Users/artemlaptev/Desktop/baza.txt"));
```

Comment:
path from the repository is always better than an absolute path - when I cloned your project to my laptop, application failed on the "File doesn't exist", because I don't have the `/Users/artemlaptev/Desktop/` directory.